src/Control/Monad/Aff.purs

```diff
@@ -65,9 +65,12 @@ module Control.Monad.Aff
   cancelWith aff c = runFn3 _cancelWith nonCanceler aff c
 
   -- | Converts the asynchronous computation into a synchronous one. All values
-  -- | and errors are ignored.
-  launchAff :: forall e a. Aff e a -> Eff e Unit
-  launchAff = runAff (const (pure unit)) (const (pure unit))
+  -- | are ignored, and if the computation produces an error, it is thrown.
+  launchAff :: forall e a. Aff e a -> Eff (err :: EXCEPTION | e) Unit
```

Comment:
The only other alternative I can see is not adding the `EXCEPTION` effect type, so as not to give the user the (false) assurance they can catch it. It's fundamentally un-catchable. Maybe that's a sign the user should always have to supply an error handler (and we could provide a global one if it makes sense?).